server/ServerNetworking.cpp

```diff
@@ -165,6 +165,24 @@ PlayerConnection::PlayerConnection(boost::asio::io_context& io_context,
     m_disconnected_callback(disconnected_callback)
 {}
 
+namespace {
+    struct AsyncCloseClosure {
+        AsyncCloseClosure(boost::optional<boost::asio::ip::tcp::socket> &&socket, int id) :
+            m_socket(std::move(socket)),
+            m_id(id)
+        {}
+
+        void operator()() {
+            TraceLogger(network) << "Asynchronously closing socket for player id " << m_id;
+            m_socket->close();
```

Comment:
since `m_socket` is an `optional` should probly check that it's not empty before using?

Reply:
No, because `m_socket` cannot be empty until destructor.

Reply:
the `socket` parameter to the constructor is an option, which could be empty...?

Reply:
No, `m_socket` isn't empty in `PlayerConnection` constructor and emptied only when move non-empty value to `AsyncCloseClosure`. It's uses optional because socket cannot be moved itself.

Reply:
That might be OK if `AsyncCloseClosure` was defined locally to where it's used, but it looks like a general purpose bit of code that takes a parameter but doesn't verify its validity before using it...

Reply:
could just do `if (m_socket) m_socket->close();` or somesuch... perhaps tweak the logger calls accordingly

Reply:
Logger calls accessed only `m_id` here.